raster_api/runtime/src/app.py

```diff
@@ -67,24 +78,36 @@ async def lifespan(app: FastAPI):
 # /mosaic - PgSTAC Mosaic titiler endpoint
 ###############################################################################
 mosaic = MosaicTilerFactory(
-    router_prefix="/mosaic",
+    router_prefix="/searches/{search_id}",
+    path_dependency=SearchIdParams,
     optional_headers=optional_headers,
     environment_dependency=settings.get_gdal_config,
     process_dependency=PostProcessParams,
     router=APIRouter(route_class=LoggerRouteHandler),
-    # add /list (default to False)
-    add_mosaic_list=settings.enable_mosaic_search,
     # add /statistics [POST] (default to False)
     add_statistics=True,
     # add /map viewer (default to False)
     add_viewer=False,
     # add /bbox [GET] and /feature  [POST] (default to False)
     add_part=True,
     colormap_dependency=ColorMapParams,
+    extensions=[
+        searchInfoExtension(),
+    ],
+)
+app.include_router(mosaic.router, prefix="/searches/{search_id}", tags=["Mosaic"])
+
+add_search_register_route(
+    app,
+    prefix="/searches",
+    # any dependency we want to validate
+    # when creating the tilejson/map links
+    tile_dependencies=[mosaic.process_dependency, mosaic.colormap_dependency],
+    tags=["Mosaic"],
 )
-app.include_router(mosaic.router, prefix="/mosaic", tags=["Mosaic"])
-# TODO
-# prefix will be replaced by `/mosaics/{search_id}` in titiler-pgstac 1.0.0
+# add /list endpoint
+add_search_list_route(app, prefix="/searches", tags=["Mosaic"])
```

Comment:
Is this the equivalent of the list registered mosaics endpoint that we were setting in the .env as `VEDA_RASTER_ENABLE_MOSAIC_SEARCH` and then 

1. Inserting in the lambda env in  [infrastructure construct](https://github.com/NASA-IMPACT/veda-backend/blob/develop/raster_api/infrastructure/construct.py#L64-L66)

2. And using the [runtime config](https://github.com/NASA-IMPACT/veda-backend/blob/develop/raster_api/runtime/src/config.py#L63) to [conditionally enable the list endpoint](https://github.com/NASA-IMPACT/veda-backend/blob/develop/raster_api/runtime/src/app.py#L76) which could potentially be expensive because it is not paginated (or it was not yet paginated at some point)? 

If so, do we want it always on now?